src/Gizmos/axisDragGizmo.ts

```diff
@@ -188,7 +188,7 @@ export class AxisDragGizmo extends Gizmo {
     protected _attachedNodeChanged(value: Nullable<Node>) {
         if (this.dragBehavior) {
             this.dragBehavior.enabled = value ? true : false;
-            this._refreshForDragBehavior(this.dragBehavior.enabled, this._coloredMaterial, this._disableMaterial);
+            this._setGizmoMeshMaterial(this.dragBehavior.enabled ? this._coloredMaterial : this._disableMaterial);

```

Comment:
LGTM!

Btw, just wondering (sort of a mini feature request), do you think it would be good if dragBehavior.enabled did not update based on attachedNodeChange? Just thinking that a developer might want to control it and not worry about it changing if a different node is attached

Reply:
I can see myself setting dragBehavior.enabled = true/false manually myself and the material will update based on that, seems like quite a useful feature

Reply:
I will do it as a second PR.